server/src/main/java/org/opensearch/index/SegmentReplicationPressureService.java

```diff
@@ -92,6 +117,8 @@ public SegmentReplicationPressureService(Settings settings, ClusterService clust
 
         this.maxAllowedStaleReplicas = MAX_ALLOWED_STALE_SHARDS.get(settings);
         clusterSettings.addSettingsUpdateConsumer(MAX_ALLOWED_STALE_SHARDS, this::setMaxAllowedStaleReplicas);
+
+        this.failStaleReplicaTask = new AsyncFailStaleReplicaTask(this, TimeValue.timeValueSeconds(30));
```

Comment:
Style nitpick, but I'd move this interval to be a constant `INTERVAL` inside `AsyncFailStaleReplicaTask` since it is never set to a value other than 30 seconds.